### ISSUE TITLE
Crash when UpdateChecker can't reach GitHub

### ISSUE THREAD

**Issue by CPKreuz** [PixiEditor/PixiEditor#166]
Simply disconnect from the internet to reproduce

```
PixiEditor crashed on 2021.03.21 at 12:32:58

-------Crash message-------
System.Net.Http.HttpRequestException: No such host is known. (raw.githubusercontent.com:443)
-----Inner exception-----
System.Net.Sockets.SocketException: No such host is known.

-------Stack trace-------
   at System.Net.Http.ConnectHelper.ConnectAsync(Func`3 callback, DnsEndPoint endPoint, HttpRequestMessage requestMessage, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.CreateHttp11ConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.GetHttpConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.SendAsyncCore(HttpRequestMessage request, HttpCompletionOption completionOption, Boolean async, Boolean emitTelemetryStartStop, CancellationToken cancellationToken)
   at PixiEditor.UpdateModule.UpdateChecker.GetUpdateIncompatibleVersionsAsync(String tag) in C:\Users\phili\source\repos\PixiEditor\PixiEditor.UpdateModule\UpdateChecker.cs:line 73
   at PixiEditor.UpdateModule.UpdateChecker.IsUpdateCompatible() in C:\Users\phili\source\repos\PixiEditor\PixiEditor.UpdateModule\UpdateChecker.cs:line 64
   at PixiEditor.ViewModels.SubViewModels.Main.UpdateViewModel.<CheckForUpdate>b__17_0() in C:\Users\phili\source\repos\PixiEditor\PixiEditor\ViewModels\SubViewModels\Main\UpdateViewModel.cs:line 62
   at PixiEditor.ViewModels.SubViewModels.Main.UpdateViewModel.CheckForUpdate() in C:\Users\phili\source\repos\PixiEditor\PixiEditor\ViewModels\SubViewModels\Main\UpdateViewModel.cs:line 59
   at PixiEditor.ViewModels.SubViewModels.Main.UpdateViewModel.Owner_OnStartupEvent(Object sender, EventArgs e) in C:\Users\phili\source\repos\PixiEditor\PixiEditor\ViewModels\SubViewModels\Main\UpdateViewModel.cs:line 91
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__140_0(Object state)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.DispatcherOperation.InvokeImpl()
   at MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at MS.Internal.CulturePreservingExecutionContext.Run(CulturePreservingExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Windows.Threading.DispatcherOperation.Invoke()
   at System.Windows.Threading.Dispatcher.ProcessQueue()
   at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
   at MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)
   at System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)
   at System.Windows.Threading.Dispatcher.PushFrame(DispatcherFrame frame)
   at System.Windows.Threading.Dispatcher.Run()
   at System.Windows.Application.RunDispatcher(Object ignore)
   at System.Windows.Application.RunInternal(Window window)
   at System.Windows.Application.Run()
   at PixiEditor.App.Main()
-----Inner exception-----
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)
   at System.Net.Sockets.Socket.<ConnectAsync>g__WaitForConnectWithCancellation|283_0(AwaitableSocketAsyncEventArgs saea, ValueTask connectTask, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.DefaultConnectAsync(SocketsHttpConnectionContext context, CancellationToken cancellationToken)
   at System.Net.Http.ConnectHelper.ConnectAsync(Func`3 callback, DnsEndPoint endPoint, HttpRequestMessage requestMessage, CancellationToken cancellationToken)
```

**Comment by NandayDev** [comment]
@CPKreuz Hey, I'd like to help fix this issue. Looks like an HttpRequestException is thrown when the host is unknown (user's connectivity is down, or, less likely, GitHub sites not reachable). In this case, the update wouldn't in any case go through successfully. I could change the signature for the method of UpdateChecker.cs
![immagine](https://user-images.githubusercontent.com/60286366/112948279-2c28ce80-9138-11eb-8e09-dcba74a1c772.png)
and make it return a **bool?** instead. So we can get a third case, null, which means "Host unreachable, stop the update procedure".
Otherwise, if you prefer, we could also create a new Enum.
In case of host unreachable, this method
![immagine](https://user-images.githubusercontent.com/60286366/112948562-84f86700-9138-11eb-858f-d2cb82c899a8.png)
will return immediately false, just after line 62.
Let me know!

**Comment by flabbet** [comment]
@NandayDev Hi, I think he already solved that in #169. You might want to check out the code there and help if needed

**Comment by NandayDev** [comment]
@flabbet Oops. misread the status. Sorry @CPKreuz !
